bears/python/requirements/PySafetyBear.py

```diff
@@ -40,14 +30,28 @@ class PySafetyBear(LocalBear):
         'Python 3 Requirements',
     }
     AUTHORS = {'Bence Nagy'}
-    REQUIREMENTS = {PipRequirement('safety', '0.5.1')}
+    REQUIREMENTS = {PipRequirement('safety', '1.8.2')}
     AUTHORS_EMAILS = {'bence@underyx.me'}
     LICENSE = 'AGPL'
     CAN_DETECT = {'Security'}
 
-    def run(self, filename, file):
+    def run(self, filename, file,
+            key: str = '',
+            db_mirror: str = '',
+            cached: bool = False,
+            ignore_ids: typed_list(str) = []):
         """
         Checks for vulnerable package versions in requirements files.
+
+        :param key:           API Key for pyup.io's vulnerability database. Can
```

Comment:
we do not store keys in `.coafile` which are checked into a repository.

If `SAFETY_API_KEY` works, that will have to be how it is provided to the bear, and core will need some enhancement in future to manage bear envvars.

If a key is mandatory, the bear needs to check that `SAFETY_API_KEY` or `db_mirror` is set in an earlier phase of the Bear configuration, so it doesnt `run()` multiple times and get the same error.

Reply:
IMO `db_mirror` setting can be there ? or I remove that also?

Reply:
Lets leave db_mirror, but default None.

And if not provided, use `Bear.download_cached_file` to fetch a copy of the database and store it in the coala bears cache

https://github.com/coala/coala-bears/blob/master/bears/java/CheckstyleBear.py#L58

or, add a dependency on https://pypi.org/project/safety-db/ , and use that dataset.
(even if the data is old; we can fix that later)